person.rb

```diff
@@ -0,0 +1,28 @@
+# main class Person
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    case of_age? || @parent_permission
+    when true
+      true
+    when false
+      false
+    end
+  end
```

Comment:
- [ ] **Required:** kindly move this method above the private keyword as this is required to be public. All methods beneath the private keyword as in your case are private.

Reply:
Thanks, done!